libc/src/stdlib/heap_limit.S

```diff
@@ -0,0 +1,2 @@
+.weak __libc_heap_limit
+.set __libc_heap_limit, LIBC_FREELIST_MALLOC_SIZE
```

Comment:
Can you define this in `freelist_malloc.cpp`? We want to avoid any unnecessary assembly in libc.

Reply:
I dodged this entirely by moving to `__libc_heap_limit` to be an address rather than a size; this removes `LIBC_FREELIST_MALLOC_SIZE` entirely. `__libc_heap_limit` would then be an intentionally undefined reference in the libc.